src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+
+      if (herbivore.health <= 0) {
+        const indexToDelete = Animal.alive.findIndex(
+          (herbivoreToDelete) => herbivoreToDelete.health <= 0,
+        );
+
+        Animal.alive.splice(indexToDelete, 1);
```

Comment:
Instead of using `findIndex` and `splice` to remove dead animals from `Animal.alive`, consider using the `filter` method to remove all animals with health less than or equal to 0. This approach is preferred according to the checklist and will ensure all dead animals are removed, not just the first one found.